sinks/kafka/driver_test.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"encoding/json"
+	"github.com/optiopay/kafka/proto"
+	"github.com/stretchr/testify/assert"
+	sink_api "k8s.io/heapster/sinks/api"
+	kube_api "k8s.io/kubernetes/pkg/api"
+	kube_api_unv "k8s.io/kubernetes/pkg/api/unversioned"
+	"testing"
+	"time"
+)
+
+type msgProducedToKafka struct {
+	message string
+}
+
+type fakeKafkaProducer struct {
+	msgs []msgProducedToKafka
+}
+
+type fakeKafkaSink struct {
+	sink_api.ExternalSink
+	fakeProducer *fakeKafkaProducer
+}
+
+func NewFakeKafkaProducer() *fakeKafkaProducer {
+	return &fakeKafkaProducer{[]msgProducedToKafka{}}
+}
+
+func (producer *fakeKafkaProducer) Produce(topic string, partition int32, messages ...*proto.Message) (offset int64, err error) {
```

Comment:
thanks @mvdan,  but I cant catch your meaning, can you tell me more clearly?  : )